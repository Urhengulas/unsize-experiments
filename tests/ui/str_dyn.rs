#![feature(ptr_metadata)]

trait Trait {
    fn dispatch_me(&self);
}

impl Trait for str {
    fn dispatch_me(&self) {}
}

unsafe impl unsizing_experiments::ConstUnsize<dyn Trait> for str {
    const TARGET_METADATA: <dyn Trait as core::ptr::Pointee>::Metadata = unimplemented!();
}

fn main() {
    let s: &dyn Trait = "".coerce_unsized();
}
