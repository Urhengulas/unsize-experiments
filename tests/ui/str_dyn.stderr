error[E0599]: no method named `coerce_unsized` found for reference `&'static str` in the current scope
  --> tests/ui/str_dyn.rs:16:28
   |
16 |     let s: &dyn Trait = "".coerce_unsized();
   |                            ^^^^^^^^^^^^^^ method not found in `&str`
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
3  | use unsizing_experiments::coerce_unsized::CoerceUnsized;
   |
